lib/Parser.js

```diff
@@ -0,0 +1,337 @@
+     * 5. Check if length-3 of string is *. If not; return false.
+     * 5. Check checksum unless ignored, if checksum doesn't compute: return false.
+     * 6. Return true
+    **/
+
+    if(!_.isString(str)) {
+      return false;
+    }
+
+    str = str.trim();
+
+    if(str === '') {
+      return false;
+    }
+
+    if(str.charAt(0) !== '$' && str.charAt(0) !== '!') {
+      return false; 
+    }
+
+    if(str.charAt(str.length - 3) !== '*') {
+      return false; 
+    }
+
+    if(this.options.ignoreChecksum !== true && this.hasValidChecksum(str) === false) {
+      return false;
+    }
+
+    return true;
+  };
+
+  Parser.prototype.transformValue = function(transform, value) {
+    switch(transform) {
+      case 'uppercase':
+        return value.toUpperCase();
+        break;
+
+      case 'lowercase': 
+        return value.toLowerCase();
+        break;
+
+      default:
+        return value; 
+        break;
+    }
+  }; 
+
+  Parser.prototype.sentencePassesFilter = function(filter, values) {
+    let field = parseInt(filter.field);
+    let match = filter.value._Data;
+    let valid = true;
+
+    if(_.isString(filter.value.transform)) {
+      match = this.transformValue(filter.value.transform.toLowerCase(), match);
+    }
+
+    switch(filter.type) {
+      case 'unless':
+        if(values[field] === match) {
+          valid = false;
+        }
+        break;
+
+      // Default is IF
+      default:
+        if(values[field] !== match) {
+          valid = false;
+        }
+        break;
+    }
+
+    return valid;
+  };
+
+  Parser.prototype.getValueFromFieldset = function(fieldset, values) {
+    let value = null;
+    let fields = [];
+
+    switch(fieldset.type) {
+      case 'magneticVariation':
+        fieldset.field.forEach((index) => {
+          index = parseInt(index);
+          fields.push(values[index]);
+        });
+
+        if(fields[0].trim() !== "") {
+          value = utils.magneticVariaton.apply(utils, fields);
+        }
+        break;
+
+      case 'datetime':
+        let time = null;
+        let date = null;
+
+        fieldset.field.forEach((field) => {
+          if(field.type === 'time') {
+            time = values[parseInt(field._Data)];
+          }
+
+          if(field.type === 'date') {
+            date = values[parseInt(field._Data)];
+          }
+        });
+
+        value = utils.timestamp(time, date);
+        break;
+
+      case 'coordinate':
+        fieldset.field.forEach((index) => {
+          index = parseInt(index);
+          fields.push(values[index]);
+        });
+
+        if(fields[0].trim() !== "") {
+          value = utils.coordinate.apply(utils, fields);
+        }
+        break;
+    }
+
+    return value;
+  };
+
+  Parser.prototype.getValue = function(config, values) {
+    if(_.isObject(config.fieldset)) {
+      return this.getValueFromFieldset(config.fieldset, values);
+    }
+
+    let value = null;
```

Comment:
I built it using the latest stable version of node (4.0.0) which is a merge between Node.js and io.js. I like the new ES6 features, but I'm not sure it's the best time (I'm having lot's of issues with binary dependencies that don't like 4.0.0)